packages/jellyfish-api-core/__tests__/category/account/sendTokensToAddress.test.ts

```diff
@@ -0,0 +1,122 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { AddressBalances, SelectionModeType } from '../../../src/category/account'
+
+describe('SendTokenToAddress', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  let from: string
+
+  async function setup (): Promise<void> {
+    from = await container.call('getnewaddress')
+    await createToken(from, 'DBTC', 200)
+    await createToken(from, 'ETH', 200)
+  }
+
+  async function createToken (address: string, symbol: string, amount: number): Promise<void> {
+    const metadata = {
+      symbol,
+      name: symbol,
+      isDAT: true,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: address
+    }
+    await container.waitForWalletBalanceGTE(101)
+    await container.call('createtoken', [metadata])
+    await container.generate(1)
+
+    await container.call('utxostoaccount', [{ [address]: '100@0' }])
+    await container.generate(1)
+
+    await container.call('minttokens', [`${amount.toString()}@${symbol}`])
+    await container.generate(1)
+  }
+
+  async function sendTokensToAddress (from: AddressBalances, to: AddressBalances, selectionMode?: SelectionModeType): Promise<string> {
+    const transactionHex = await client.account.sendTokensToAddress(from, to, selectionMode)
+    await container.generate(1)
+    return transactionHex
+  }
```

Comment:
I am quite against this kind of test writing convention. This is because you are testing this function hence it should not be abstracted and be written out in detail.

While for the `createToken` that makes sense because it's part of the setup process and not really what you are testing for. You are testing `client.account.sendTokensToAddress` so I expect this to be in the test body `it('')`.

Reply:
Understood. Actually jimmy and i talked about it. Let me refactor it.